crates/vm/levm/src/vm.rs

```diff
@@ -155,7 +158,11 @@ impl VM {
 
         // Add precompiled contracts addresses to cache.
         // TODO: Use the addresses from precompiles.rs in a future
-        let max_precompile_address = if env.spec_id >= SpecId::CANCUN { 10 } else { 9 };
+        let max_precompile_address = match env.spec_id {
+            spec if spec >= SpecId::PRAGUE => SIZE_PRECOMPILES_PRAGUE,
+            spec if spec >= SpecId::CANCUN => SIZE_PRECOMPILES_CANCUN,
+            _ => SIZE_PRECOMPILES_PRE_CANCUN,
```

Comment:
Nit. It is always better to explicit the "else" case of the `match`.
```suggestion
            spec if spec < SpecId::CANCUN => SIZE_PRECOMPILES_PRE_CANCUN,
```

Reply:
The compiler seems to need us to add the "else" case. Let's add the `_` case and return an internal error in this case (this should be an unreachable case).

Reply:
Done, [49a699b](https://github.com/lambdaclass/ethrex/pull/1691/commits/49a699b4f1f4d54feab8c1f50e5674b5cf004910)